### PR TITLE
Optimize OperanceDataController with lazy loading for expanded rows

### PR DESCRIPTION
**Pull request by fjyeo**
- Implement lazy loading and caching of expanded row data
- Add expandedRowFetcher to initialize method
- Update toggleExpandedRow to fetch data on demand
- Add getExpandedRowData method for accessing expanded data
- Include clearUnusedExpandedData for memory management

### CODE REVIEW

File: lib/src/operance_data_controller.dart [hunk 1]
```diff
@@ -310,12 +319,32 @@ class OperanceDataController<T> extends ChangeNotifier {
...
+
+    if (!isExpanded && _expandedRowFetcher != null) {
+      if (!_expandedRowData.containsKey(index)) {
+        try {
+          _expandedRowData[index] = Future.value(
+            await _expandedRowFetcher!(allRows[index]),
+          );
+        } on Exception {
+          _expandedRows[index] = false;
+          _expandedRowData.remove(index);
```
Comment (fjyeo) [inline lib/src/operance_data_controller.dart:335]: not sure how to sort slight issue with the error handling. Maybe log the error and notify the users

**Comment by zxcvbnmmohd** [comment]
why does the expanded rows need to be lazy loaded? The expandedWidget builder is just something that a developer can use to show more data for a row that's already loaded. It's not something that'll be preloaded before the actual items are loaded. What use case would this be beneficial for?

Also this change is only implemented in the controller and no where else so it won't actually affect anything really in terms of the expanded row builder itself.

File: lib/src/operance_data_controller.dart (hunk 1 above)
Comment (zxcvbnmmohd) [reply lib/src/operance_data_controller.dart:335]: the devs should always handle their own exceptions, we shouldn't handle it for them

**Review by zxcvbnmmohd — CHANGES_REQUESTED** [review]
Just a few questions

**Comment by fjyeo** [comment]
Completely my fault here, I misunderstood the purpose and implementation of the expanded rows in our OperanceDataTable and thought that the lazy loading would be beneficial so no preloaded data will be used so the code changes here are not useful.

**Pull request closed without merging**